### PR TITLE
Hashicorp Vault Fixes

### PR DESCRIPTION
**Pull request by liorblob**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/16317

Support KV 1 engine in `list-secrets` command
Fix non-indicative test errors
Fix multiple engine configurations

### CODE REVIEW

File: Integrations/integration-HashiCorp-Vault.yml [hunk 1]
```diff
@@ -602,12 +612,15 @@ script:
 
         for engine_type in ENGINES:
             engines_to_fetch = list(filter(lambda e: e['type'] == engine_type, ENGINE_CONFIGS))
-            if len(engines_to_fetch) == 0:
-                return_error('Engine type not configured, please use the configure-engine command to configure a secrets engine.')
             engines_to_fetch_from += engines_to_fetch
 
+        if len(engines_to_fetch_from) == 0:
+                return_error('Engine type not configured, please use the configure-engine command to configure a secrets engine.')
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: remove `please`

File: Integrations/integration-HashiCorp-Vault.yml [hunk 2]
```diff
@@ -602,12 +612,15 @@ script:
...
+
         for engine in engines_to_fetch_from:
             if engine['type'] == 'KV':
+                if 'version' not in engine:
+                    return_error('Version not configured for KV engine, please re-configure')
```
Comment (Itay4) [inline Integrations/integration-HashiCorp-Vault.yml]: remove `please`

**Comment by yardensade** [comment]
What are we going to do about the missing descriptions?

**Comment by liorblob** [comment]
@yardensade some copy pasting is needed

**Pull request merged**